tests/integration/extenders/FormatterTest.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Tests\integration\extenders;
+
+use Flarum\Extend;
+use Flarum\Formatter\Formatter;
+use Flarum\Tests\integration\TestCase;
+
+class FormatterTest extends TestCase
+{
+    protected function getFormatter()
+    {
+        return $this->app()->getContainer()->make(Formatter::class);
+    }
+
+    /**
+     * @test
+     */
+    public function custom_formatter_config_doesnt_work_by_default()
+    {
+        $formatter = $this->getFormatter();
+        $formatter->flush();
+
+        $this->assertEquals('<t>[B]something[/B]</t>', $formatter->parse('[B]something[/B]'));
+    }
+
+    /**
+     * @test
+     */
+    public function custom_formatter_config_works_if_added_with_closure()
+    {
+        $this->extend((new Extend\Formatter)->configure(function ($config) {
+            $config->BBCodes->addFromRepository('B');
+        }));
+
+        // Needed so that the config is refreshed, this is usually handled by lifecyclehooks.
+        $formatter = $this->getFormatter();
+        $formatter->flush();
```

Comment:
Can you expand on why this is necessary? This sounds like state from one test is leaking over to the next. We should try to get each test to always start with a completely flushed formatter (i.e. cache in this case, it seems).